### PR TITLE
ci: Add an ASAN context

### PR DESCRIPTION
**Pull request by cgwalters**
Let's start doing this by default.  At some point, I'd like
to fold all of the sanitizers into one run or so.

### CODE REVIEW

File: .redhat-ci.Dockerfile [hunk 1]
```diff
@@ -10,7 +10,7 @@ RUN dnf install -y \
         gjs \
         parallel \
         clang \
-        libubsan \
+        lib{ub,a,t}san \
```
Comment (jlebon) [inline .redhat-ci.Dockerfile]: This isn't going to take effect until it's part of the tree, so you'll also have to explicitly list the new packages in the YAML until then. I suppose that's probably why the `cc` sanity check is failing.

**Comment by cgwalters** [comment]
:question: :bomb: `checking for GLIB - version >= 2.0.0... no`

**Comment by cgwalters** [comment]
bot, retest this please

**Comment by jlebon** [comment]
Yeah, clearly, we need to provide better feedback when validation fails. In this case, the issue is that the ASAN suite doesn't have a `build` section, so the Clang suite inherits neither a `build` nor a `tests` section, which is invalid.

**Comment by cgwalters** [comment]
bot, retest this please

File: .redhat-ci.yml [hunk 2]
```diff
@@ -52,9 +52,9 @@ packages:
 
 tests:
     - env NOCONFIGURE=1 ./autogen.sh
-    # Suppress leak detection in configure, because sadly some of them leak
+    # Suppress leak detection in configure (and build, since g-ir-scanner uses glib), because sadly some of them leak
     - env CFLAGS='-fsanitize=address' ASAN_OPTIONS=detect_leaks=false ./configure --prefix=/usr --libdir=/usr/lib64 --disable-gtk-doc
-    - make -j $(getconf _NPROCESSORS_ONLN)
+    - ASAN_OPTIONS=detect_leaks=false make -j $(getconf _NPROCESSORS_ONLN)
```
Comment (jlebon) [inline .redhat-ci.yml]: So then can't we go back to the previous format and just override `env`?

**Comment by jlebon** [comment]
Is this ready to merge, or did you mean to also use this PR to fix the leaks? Judging by the logs, that might take a while. :) (Should we open an issue to track this work, maybe?)

**Comment by cgwalters** [comment]
I don't understand what's leaking.  I'm not reproducing it locally - I was originally using a f25 container with some `sudo make install` in it, then I tried a f24 container, now I'm trying to use the exact container being tested here.

**Comment by cgwalters** [comment]
Okay, still getting leaks in CI env that I'm not seeing locally.  Will investigate in background.

**Comment by cgwalters** [comment]
Obsoleted by https://github.com/ostreedev/ostree/pull/622

**Pull request closed without merging**